test/jdk/java/awt/Graphics2D/CopyAreaOOB.java

```diff
@@ -50,73 +106,50 @@ public void paint(Graphics g) {
-        testRegion(capture, "original red",   0,  10,  50, 400, 0xffff0000);
-        testRegion(capture, "background",    50,  10,  60, 400, 0xff000000);
-        testRegion(capture, "in-between",    60,  10, 110,  20, 0xff000000);
-        testRegion(capture, "copied red",    60,  20, 110, 400, 0xffff0000);
-        testRegion(capture, "background",   110,  10, 400, 400, 0xff000000);
+        g2d.copyArea(0, 10, 50, h - 10, 60, 10);
+        g2d.dispose();
     }
 
     public Dimension getPreferredSize() {
         return new Dimension(400, 400);
     }
 
-    private static void testRegion(BufferedImage bi, String name,
+    private static void testRegion(String region,
                                    int x1, int y1, int x2, int y2,
-                                   int expected)
-    {
+                                   int expected) {
+        System.out.print("Testing region: " + region);
         for (int y = y1; y < y2; y++) {
             for (int x = x1; x < x2; x++) {
-                int actual = bi.getRGB(x, y);
+                int actual = captureImg.getRGB(x, y);
                 if (actual != expected) {
-                    throw new RuntimeException("Test failed for " + name +
-                                                       " region at x="+x+" y="+y+
-                                                       " (expected="+
-                                                       Integer.toHexString(expected) +
-                                                       " actual="+
-                                                       Integer.toHexString(actual) +
-                                                       ")");
+                    System.out.print(" Status: FAILED\n");
+                    errorLog.append("Test failed for " + region
+                                               + " region at x: " + x + " y: " + y
+                                               + " (expected: "
+                                               + Integer.toHexString(expected)
+                                               + " actual: "
+                                               + Integer.toHexString(actual) + ")\n");
+                    return;
                 }
             }
         }
+        System.out.print(" Status: PASSED\n");
     }
 
-    public static void main(String[] args) {
-        boolean show = (args.length == 1) && ("-show".equals(args[0]));
-
-        CopyAreaOOB test = new CopyAreaOOB();
-        Frame frame = new Frame();
-        frame.setUndecorated(true);
-        frame.add(test);
-        frame.pack();
-        frame.setLocationRelativeTo(null);
-        frame.setVisible(true);
-
+    private static void saveImages() {
+        GraphicsConfiguration ge = GraphicsEnvironment
+                .getLocalGraphicsEnvironment().getDefaultScreenDevice()
+                .getDefaultConfiguration();
+        BufferedImage screenCapture = robot.createScreenCapture(ge.getBounds());
```

Comment:
Using a multi-resolution screen capture could shed some light on failures… If Mac uses a Retina-like display, the captured image is scaled down, which affects the colors on the image.